docs/core/diagnostics/sos-debugging-extension.md

```diff
@@ -0,0 +1,367 @@
+---
+title: "SOS Debugging Extension for .NET"
+description: Provides information about the internal CLR environment.
+ms.date: "12/21/2020"
+helpviewer_keywords:
+  - "debugging extensions"
+  - "SOS debugging extensions"
+  - "SOS.dll"
+---
+
+# SOS debugging extension
+
+The SOS Debugging Extension lets you view information about code that is running inside the .NET Core runtime, both on live processes and dumps. The extension is pre-installed with [dotnet-dump](dotnet-dump.md) and [Windbg/dbg](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/debugger-download-tools), and can be [downloaded](dotnet-sos.md) for use with LLDB.  The SOS Debugging Extension is useful for collecting information about the managed heap, look for heap corruptions, display internal data types used by the runtime, and view information about all managed code running inside the runtime.
+
+## Syntax
+
+### Windows
+
+`![command] [options]`
+
+### Linux and MacOS
+
+`sos [command] [options]`
+
+Many of the commands have aliases or short cuts under lldb:
+
+`clrstack [options]`
+
+## Commands
+
+|Command|Description|
+|-------------|-----------------|
+|**bpmd** [**-nofuturemodule**] [\<*module name*> \<*method name*>] [**-md** <`MethodDesc`>] **-list** **-clear** \<*pending breakpoint number*> **-clearall**|Creates a breakpoint at the specified method in the specified module.<br /><br /> If the specified module and method have not been loaded, this command waits for a notification that the module was loaded and just-in-time (JIT) compiled before creating a breakpoint.<br /><br /> You can manage the list of pending breakpoints by using the **-list**, **-clear**, and **-clearall** options:<br /><br /> The **-list** option generates a list of all the pending breakpoints. If a pending breakpoint has a non-zero module ID, that breakpoint is specific to a function in that particular loaded module. If the pending breakpoint has a zero module ID, that breakpoint applies to modules that have not yet been loaded.<br /><br /> Use the **-clear** or **-clearall** option to remove pending breakpoints from the list.|
```

Comment:
Personally I would rather teach how to get help from the command line than duplicating the command line help into the docs.  I am not sure it is adding value.

The formatting of this section seems crude.  I find it hard to read.  But perhaps it is because my monitors are in portrait layout instead of landscape.

I think this isn't consistent with the style guide.  However adding a `Synopsis` for every command could get very verbose.

Reply:
I think it's still fine to have this as a reference document on SOS, which you can't find anywhere on the Internet other than our repo currently. It'll be more easily indexed and searchable by users. 

I'll leave the docs team to decide whether this table should be separated out into commands and add synopsis on each of them, but if that is indeed what needs to happen, I'd be happy to make the changes if @tommcdon wants to go write some other docs first :p.

Reply:
If these are commands via command line, they should follow a similar style of https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-publish but we probably don't need a single page for each command. Getting away from the table style makes it easier to write though.

Reply:
I'm going to leave the table as-is for now and wait for customer feedback.